inc/Modules/MediaLibrary/ConsumerAdmin.php

```diff
@@ -74,23 +64,25 @@ public function setup_hooks(): void {
 	public function add_sync_filter(): void {
 		global $pagenow;
 
-		if ( 'upload.php' === $pagenow ) {
-			// Nonce verification for filter form.
-			$nonce = filter_input( INPUT_GET, 'onemedia_sync_nonce', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
-			$nonce = isset( $nonce ) ? sanitize_text_field( wp_unslash( $nonce ) ) : '';
+		if ( 'upload.php' !== $pagenow ) {
+			return;
+		}
 
-			if ( ! $nonce ) {
-				// This means this is the first load of the page, so we don't have onemedia_sync_filter nonce yet.
-				echo onemedia_get_template_content( 'brand-site/sync-status' ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
-			} elseif ( $nonce && wp_verify_nonce( $nonce, 'onemedia_sync_filter' ) ) {
-				// This means the form has been submitted, so we have a nonce to verify.
-				$sync_status = isset( $_GET[ Constants::ONEMEDIA_SYNC_STATUS_POSTMETA_KEY ] )
-					? sanitize_text_field( wp_unslash( $_GET[ Constants::ONEMEDIA_SYNC_STATUS_POSTMETA_KEY ] ) )
-					: '';
-
-				// Escaping handled in the template file.
-				echo onemedia_get_template_content( 'brand-site/sync-status', array( 'sync_status' => $sync_status ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
-			}
+		// Nonce verification for filter form.
+		$nonce = filter_input( INPUT_GET, 'onemedia_sync_nonce', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		$nonce = isset( $nonce ) ? sanitize_text_field( wp_unslash( $nonce ) ) : '';
+
+		if ( ! $nonce ) {
+			// This means this is the first load of the page, so we don't have onemedia_sync_filter nonce yet.
+			echo MediaSharingAdmin::onemedia_get_template_content( 'brand-site/sync-status' ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		} elseif ( $nonce && wp_verify_nonce( $nonce, 'onemedia_sync_filter' ) ) {
+			// This means the form has been submitted, so we have a nonce to verify.
+			$sync_status = isset( $_GET[ Media_Sharing_Controller::ONEMEDIA_SYNC_STATUS_POSTMETA_KEY ] )
```

Comment:
The model is backwards. Rest Controllers wrap _plugin_ logic, they shouldn't contained the shared plugin logic directly. 

Between this and the ::get_template_content()`, Might make sense to create a MediaLibrary/Utils.php for the shared (public static) module logic.